pkg/sources/job_report.go

```diff
@@ -1,37 +1,269 @@
 package sources
 
+//go:generate mockgen --source=./job_report.go --destination=mock_job_report_test.go --package=sources
```

Comment:
what's this?

Reply:
Go supports running commands marked by this directive to generate *stuff*.

In this case it is running the command [mockgen](https://github.com/uber-go/mock) which will create mocks of all the interfaces in the source file and write it to the destination file.

To generate that file, I ran `go generate ./pkg/sources/...`.

More info: [go:generate](https://www.practical-go-lessons.com/post/what-is-go-generate-and-how-to-use-it-ccava8v5toqc70ipi1jg)

I chose to use it because the interface was large and I really just wanted to test whether the methods got called with the expected values, which `mockgen` supports.